packages/the-org/src/communityManager/index.ts

```diff
@@ -403,6 +405,133 @@ const config: OnboardingConfig = {
+    {
+      name: 'Test Conflict Resolution',
+      fn: async (runtime: IAgentRuntime) => {
+        this.scenarioService = runtime.getService('scenario');
+        if (!this.scenarioService) {
+          throw new Error('Scenario service not found');
+        }
+
+        const worldId = await this.scenarioService.createWorld('Conflict Test', 'Test Owner');
+        const roomId = await this.scenarioService.createRoom(worldId, 'general');
+
+        await this.scenarioService.addParticipant(worldId, roomId, runtime.agentId);
+        const userId = uuidv4() as UUID;
+        await this.scenarioService.addParticipant(worldId, roomId, userId);
+
+        await this.scenarioService.sendMessage(
+          runtime,
+          worldId,
+          roomId,
+          "There's a user causing disruptions in the general channel"
+        );
+
+        const completed = await this.scenarioService.waitForCompletion(8000);
+        if (!completed) {
+          throw new Error('Agent did not resolve conflict in time');
+        }
+      }
+    },
+    {
+      name: 'Test New User Onboarding',
+      fn: async (runtime: IAgentRuntime) => {
+        this.scenarioService = runtime.getService('scenario');
+        if (!this.scenarioService) {
+          throw new Error('Scenario service not found');
+        }
+
+        const worldId = await this.scenarioService.createWorld('Onboarding Test', 'Test Owner');
+        const roomId = await this.scenarioService.createRoom(worldId, 'welcome');
+
+        await this.scenarioService.addParticipant(worldId, roomId, runtime.agentId);
+        const newUserId = uuidv4() as UUID;
+        await this.scenarioService.addParticipant(worldId, roomId, newUserId);
+
+        await this.scenarioService.sendMessage(
+          runtime,
+          worldId,
+          roomId,
+          "Hi everyone, I'm new here!"
+        );
+
+        const completed = await this.scenarioService.waitForCompletion(5000);
+        if (!completed) {
+          throw new Error('Agent did not complete onboarding in time');
+        }
+      }
+    },
+    {
+      name: 'Test Moderation Actions',
+      fn: async (runtime: IAgentRuntime) => {
+        this.scenarioService = runtime.getService('scenario');
+        if (!this.scenarioService) {
+          throw new Error('Scenario service not found');
+        }
+
+        const worldId = await this.scenarioService.createWorld('Moderation Test', 'Test Owner');
+        const roomId = await this.scenarioService.createRoom(worldId, 'moderation');
+
+        await this.scenarioService.addParticipant(worldId, roomId, runtime.agentId);
+        const userId = uuidv4() as UUID;
+        await this.scenarioService.addParticipant(worldId, roomId, userId);
+
+        await this.scenarioService.sendMessage(
+          runtime,
+          worldId,
+          roomId,
+          "This user posted inappropriate content"
+        );
+
+        const completed = await this.scenarioService.waitForCompletion(10000);
```

Comment:
can we make the waitForCompletion time the same across test cases and make it a variable that is configurable